controllers/sriovoperatorconfig_controller.go

```diff
@@ -81,7 +83,9 @@ func (r *SriovOperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			logger.Info("default SriovOperatorConfig object not found. waiting for creation.")
-			return reconcile.Result{}, nil
+
+			err := r.deleteAllWebhooks(ctx)
+			return reconcile.Result{}, err
```

Comment:
Why don't we use owner reference? Then, all of this is automatically handled.

Reply:
Because ValidatingWebhookConfiguration and MutatingWebhookConfiguration resources are not namespaced, while the OwnerReference field works with resources in the same namespace as the owner.